packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -210,11 +210,14 @@ class CurlyComponentManager {
 
     // We usually do this in the `didCreateElement`, but that hook doesn't fire for tagless components
     if (component.tagName === '') {
+      if (environment.isInteractive) {
+        component.trigger('willRender');
+      }
```

Comment:
I think you need to move this into `create`? I thought the point is you want to `set` on the component before we apply the bindings